colorfield/fields.py

```diff
@@ -12,28 +12,51 @@
     from django.utils.translation import ugettext_lazy as _
 
 
-COLOR_RE = re.compile('^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$')
-color_validator = RegexValidator(
-    COLOR_RE, _('Enter a valid color.'), 'invalid')
+COLOR_HEX_RE = re.compile('^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$')
+color_hex_validator = RegexValidator(
+    COLOR_HEX_RE, _('Enter a valid color.'), 'invalid')
+
+COLOR_HEXA_RE = re.compile('#([A-Fa-f0-9]{8}|[A-Fa-f0-9]{4})$')
+color_hexa_validator = RegexValidator(
+    COLOR_HEXA_RE, _('Enter a valid color.'), 'invalid')
+
+VALIDATORS_PER_FORMAT = {
+    'hex': color_hex_validator,
+    'hexa': color_hexa_validator
+}
+
+WHITE_IN_FORMAT = {
+    'hex': '#FFFFFF',
+    'hexa': '#FFFFFFFF'
+}
 
 
 class ColorField(models.CharField):
 
-    default_validators = [color_validator]
+    default_validators = []
 
     def __init__(self, *args, **kwargs):
+        color_format = kwargs.pop('color_format', 'hex')
+        self.palette = kwargs.pop('palette', [])
+        if color_format not in ['hex', 'hexa']:
+            raise ValueError('Unsupported color format: {}'.format(color_format))
+        self.color_format = color_format
+        self.default_validators = [VALIDATORS_PER_FORMAT[self.color_format]]
+
         kwargs.setdefault('max_length', 18)
         if kwargs.get('null'):
             kwargs.setdefault('blank', True)
             kwargs.setdefault('default', None)
         elif kwargs.get('blank'):
             kwargs.setdefault('default', '')
         else:
-            kwargs.setdefault('default', '#FFFFFF')
+            kwargs.setdefault('default', WHITE_IN_FORMAT[self.color_format])
         super(ColorField, self).__init__(*args, **kwargs)
 
     def formfield(self, **kwargs):
         kwargs['widget'] = ColorWidget(attrs={
             'default': self.get_default(),
+            'color_format': self.color_format,
+            'palette': self.palette,
```

Comment:
I think that if there is a **palette** defined, the spectrum for choosing the color should be hidden (I think there is a `jscolor` option for this). 

Generally a palette is used to force the user to choose only a predefined color without the possibility to choose a custom one.

Reply:
The use case for my company is actually to have both at the same time.

Regardless, I haven't been able to found an option in jscolor. It can be probably still be achieved with some JS but that's not my forte...
Also it would probably make sense in this case to dynamically generate a validator for the field that enforce that value is in the palette.

Reply:
I understand, it would be nice to add another option `palette_only` to hide the spectrum in case there is a `palette` available. What do you think about it?

To hide the spectrum maybe it is possible to do it by just set `width` and `height` options to `0`:
https://jscolor.com/docs/#api--opts--width

Reply:
I have tried settings width and height but it's not enough. sliders and spectrum pointer are still on screen.
![image](https://user-images.githubusercontent.com/7900305/104487054-e8103e00-55cc-11eb-82d4-07abfd806872.png)

Reply:
Rather than using palette_only, It could be nice to make it works with the field `choices` directly

Reply:
How using `choices`?

Reply:
The same way it already work for regular CharField (https://docs.djangoproject.com/en/3.1/ref/models/fields/#choices
). 
When you set choices on a Charfield, it changes the default widget from a TextInput to a Select with the different choices.

```python

CHOICES = [
    ('#FFFFFF', 'white'),
    ('#000000', 'black')
]
color = ColorField(choices=CHOICES)
```

For the color field you would:
- set the palette to different choices
- set a bool to hide the spectrum

Reply:
Ahhh yes, it would be great to convert automatically `choices` option to `jscolor` palette.
Could you do this before merging this PR?

Reply:
done